src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,728 @@
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        if (tmhPatientData.DiagnosticReportAsImages != null &&
+                            tmhPatientData.DiagnosticReportAsImages.Any())
+                        {
+                            var diagnosticReportResponse = FetchDiagnosticReportImagesData(dataRequest,
+                                tmhPatientData.DiagnosticReportAsImages, patientName).GetAwaiter().GetResult();
+                            if (diagnosticReportResponse.HasValue)
+                            {
+                                var serializeObject = JsonConvert.SerializeObject(
+                                    diagnosticReportResponse.ValueOr((DiagnosticReportResponse) null),
+                                    new JsonSerializerSettings
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        break;
+                    }
+                    case HiType.Medication:
+                        break;
+                    case HiType.DocumentReference:
+                        break;
+                    case HiType.Prescription:
+                        break;
+                    case HiType.DischargeSummary:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            var entries = new Entries(careBundles);
+            return Option.Some(entries);
+        }
+
+        private static async Task<Option<DiagnosticReportResponse>> FetchDiagnosticReportImagesData(
+            DataRequest dataRequest, List<DiagnosticReportAsImage> diagnosticReportAsImages, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var uuid = Uuid.Generate().Value;
+            var id = uuid.Split(":").Last();
+            var representations = new List<IDiagnosticReport>();
+            foreach (var reportAsImage in diagnosticReportAsImages)
+            {
+                if (!WithinRange(dataRequest.DateRange, reportAsImage.Issued)) continue;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)